cmake/FindCMSIS.cmake

```diff
@@ -178,10 +178,19 @@ ENDFOREACH()
 IF(STM32_CHIP_TYPE)
     SET(CMSIS_STARTUP_SOURCE_FILE SRC_FILE-NOTFOUND)
     FIND_FILE(CMSIS_STARTUP_SOURCE_FILE ${CMSIS_STARTUP_SOURCE}
-        PATH_SUFFIXES src stm32${STM32_FAMILY_LOWER} cmsis
-        HINTS ${STM32Cube_DIR}/Drivers/CMSIS/Device/ST/STM32${STM32_FAMILY}xx/Source/Templates/gcc/
+        PATH_SUFFIXES src Src SRC
+        HINTS ${PROJECT_SOURCE_DIR}
         CMAKE_FIND_ROOT_PATH_BOTH
     )
+    IF(NOT EXISTS ${CMSIS_STARTUP_SOURCE_FILE})
+        FIND_FILE(CMSIS_STARTUP_SOURCE_FILE ${CMSIS_STARTUP_SOURCE}
+            PATH_SUFFIXES src stm32${STM32_FAMILY_LOWER} cmsis
+            HINTS ${STM32Cube_DIR}/Drivers/CMSIS/Device/ST/STM32${STM32_FAMILY}xx/Source/Templates/gcc/
+            CMAKE_FIND_ROOT_PATH_BOTH
+        )
+    ELSE()
+        MESSAGE(STATUS "CMSIS startup found locally")
```

Comment:
The message is a good idea, can't you add it in all cases and give the path. This way the user knows what's goins on